components/playlists/browser/playlists_media_file_controller.cc

```diff
@@ -0,0 +1,342 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/playlists/browser/playlists_media_file_controller.h"
+
+#include <utility>
+
+#include "base/bind.h"
+#include "base/files/file.h"
+#include "base/files/file_util.h"
+#include "base/json/json_reader.h"
+#include "base/json/json_writer.h"
+#include "base/strings/string_number_conversions.h"
+#include "base/task/post_task.h"
+#include "base/task_runner_util.h"
+#include "brave/components/playlists/browser/playlists_types.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "services/network/public/cpp/simple_url_loader.h"
+#include "url/gurl.h"
+
+namespace {
+
+constexpr unsigned int kRetriesCountOnNetworkChange = 1;
+const base::FilePath::StringType kSourceMediaFilesDir(
+    FILE_PATH_LITERAL("source_files"));
+const base::FilePath::StringType kUnifiedMediaFileName(
+    FILE_PATH_LITERAL("media_file"));
+
+net::NetworkTrafficAnnotationTag GetNetworkTrafficAnnotationTagForURLLoad() {
+  return net::DefineNetworkTrafficAnnotation("playlists_controller", R"(
+      semantics {
+        sender: "Brave Playlists Controller"
+        description:
+          "Fetching thumbnail image for newly created playlist"
```

Comment:
copy/paste error? should say fetching media file?

Reply:
fixed.